### PR TITLE
Merge pull request #608 from Janig88/teenmegaworld-fix

### PR DESCRIPTION
**Pull request by Janig88**
Should now pull all correct info, except i could not get pulling poster to work

### CODE REVIEW

File: Contents/Code/networkTMW.py [hunk 1]
```diff
@@ -25,71 +25,49 @@ def update(metadata,siteID,movieGenres,movieActors):
     urlBase = PAsearchSites.getSearchBaseURL(siteID)
 
     # Title
-    metadata.title = detailsPageElements.xpath('//section[@class="video-block"]//h1')[0].text_content().strip()
+    metadata.title = detailsPageElements.xpath('//div[@class="title-line"]/h1')[0].text_content().strip()
     Log("Scene Title: " + metadata.title)
```
Comment (claygoldfinch) [inline Contents/Code/networkTMW.py]: Remove

File: Contents/Code/networkTMW.py [hunk 2]
```diff
@@ -25,71 +25,49 @@ def update(metadata,siteID,movieGenres,movieActors):
...
     try:
-        metadata.summary = detailsPageElements.xpath('//p[@class="description"]')[0].text_content().strip()
+        metadata.summary = detailsPageElements.xpath('//div[@class="text"]//p')[0].text_content().strip()
     except:
         Log('No summary found')
 
     # Date
-    date = detailsPageElements.xpath('//section[@class="video-block"]//time')[0].text_content().strip()
+    date = detailsPageElements.xpath('//div[@class="date"]//time')[0].text_content().strip()
     Log("date: " + date)
```
Comment (claygoldfinch) [inline Contents/Code/networkTMW.py]: Remove

File: Contents/Code/networkTMW.py [hunk 3]
```diff
@@ -25,71 +25,49 @@ def update(metadata,siteID,movieGenres,movieActors):
...
     # Actors
     movieActors.clearActors()
-    actors = detailsPageElements.xpath('//ul[@class="girls"]//a')
+    actors = detailsPageElements.xpath('//div[@class="video"]//div[@class="site"]//a[position()>1]')
     Log("actors #: " + str(len(actors)))
     if len(actors) > 0:
         for actorLink in actors:
             actorName = actorLink.text_content().strip()
             Log("Actor: " + actorName)
             actorPageURL = actorLink.get('href').replace('https','http')
```
Comment (claygoldfinch) [inline Contents/Code/networkTMW.py]: Remove ".replace('https','http')"

File: Contents/Code/networkTMW.py [hunk 4]
```diff
@@ -25,71 +25,49 @@ def update(metadata,siteID,movieGenres,movieActors):
...
             actorPageElements = HTML.ElementFromURL(actorPageURL)
-            actorPhotoURL = urlBase + actorPageElements.xpath('//div[@class="information"]//img')[0].get("src").replace('https','http')
+            actorPhotoURL = urlBase + actorPageElements.xpath('//div[@class="photo"]//img')[0].get("data-src").replace('https','http')
```
Comment (claygoldfinch) [inline Contents/Code/networkTMW.py]: Replace "urlBase" with "PAsearchSites.getSearchBaseURL(siteID)"

File: Contents/Code/networkTMW.py [hunk 5]
```diff
@@ -25,71 +25,49 @@ def update(metadata,siteID,movieGenres,movieActors):
...
             Log("ActorPhotoURL: " + actorPhotoURL)
```
Comment (claygoldfinch) [inline Contents/Code/networkTMW.py]: Remove

File: Contents/Code/networkTMW.py [hunk 6]
```diff
@@ -25,71 +25,49 @@ def update(metadata,siteID,movieGenres,movieActors):
     urlBase = PAsearchSites.getSearchBaseURL(siteID)
```
Comment (claygoldfinch) [inline Contents/Code/networkTMW.py]: Remove and add "art = []"

File: Contents/Code/networkTMW.py [hunk 7]
```diff
@@ -25,71 +25,49 @@ def update(metadata,siteID,movieGenres,movieActors):
...
 
     # Actors
     movieActors.clearActors()
-    actors = detailsPageElements.xpath('//ul[@class="girls"]//a')
+    actors = detailsPageElements.xpath('//div[@class="video"]//div[@class="site"]//a[position()>1]')
     Log("actors #: " + str(len(actors)))
     if len(actors) > 0:
         for actorLink in actors:
             actorName = actorLink.text_content().strip()
             Log("Actor: " + actorName)
```
Comment (claygoldfinch) [inline Contents/Code/networkTMW.py]: Remove

File: Contents/Code/networkTMW.py [hunk 8]
```diff
@@ -25,71 +25,49 @@ def update(metadata,siteID,movieGenres,movieActors):
...
     # Genres
     movieGenres.clearGenres()
-    genres = detailsPageElements.xpath('//ul[@class="tags"]//a')
+    genres = detailsPageElements.xpath('//ul[@class="tag-list"]//a')
     if len(genres) > 0:
         for genreLink in genres:
             genre = genreLink.text_content()
             movieGenres.addGenre(genre)
 
-    # Posters/Background
```
Comment (claygoldfinch) [inline Contents/Code/networkTMW.py]: No artwork will be scraped if this is removed. Add this:

```
    ### Posters and artwork ###

    # Video trailer background image
    try:
        twitterBG = PAsearchSites.getSearchBaseURL(siteID) + detailsPageElements.xpath('//video')[0].get('poster')
        art.append(twitterBG)
    except:
        pass

    j = 1
    Log("Artwork found: " + str(len(art)))
    for posterUrl in art:
        if not PAsearchSites.posterAlreadyExists(posterUrl,metadata):
            metadata.posters[posterUrl] = Proxy.Preview(HTTP.Request(posterUrl, headers={'Referer': 'http://www.google.com'}).content, sort_order = j)
            metadata.art[posterUrl] = Proxy.Preview(HTTP.Request(posterUrl, headers={'Referer': 'http://www.google.com'}).content, sort_order = j)
            j = j + 1
```

File: Contents/Code/networkTMW.py [hunk 9]
```diff
@@ -23,73 +23,60 @@ def update(metadata,siteID,movieGenres,movieActors):
...
-    valid_names = list()
-    metadata.posters.validate_keys(valid_names)
-    metadata.art.validate_keys(valid_names)
+    ### Posters and artwork ###
 
+    # Video trailer background image
     try:
-        background = urlBase + detailsPageElements.xpath('//div[@class="media-wrapper"]//video')[0].get("poster").replace('https','http')
+        twitterBG = PAsearchSites.getSearchBaseURL(siteID) + detailsPageElements.xpath('//video')[0].get('poster')
```
Comment (claygoldfinch) [inline Contents/Code/networkTMW.py]: Update to

```
    try:
        twitterBG = PAsearchSites.getSearchBaseURL(siteID) + detailsPageElements.xpath('//video')[0].get('poster')
        art.append(twitterBG)
    except:
        twitterBG = PAsearchSites.getSearchBaseURL(siteID) + detailsPageElements.xpath('//deo-video')[0].get('cover-image')
        art.append(twitterBG)
```

**Comment by claygoldfinch** [comment]
> 
> 
> I'm not sure i understood correctly
> "Figured we could remove some of the pre-existing verbose logging".
> logging now removed and scraping poster now works, thanks for the suggestion.

Yep, you got it - just wanted all the logging removed.

Made a few more suggestions, fix those up and I think we're good to merge!

**Comment by Janig88** [comment]
Done, does it look alright now.

**Pull request merged**